### PR TITLE
implementation of the fetch api

### PR DESCRIPTION
**Pull request by paweldowgiallo84**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
The task was to get all existing flashcards from the database when initializing the app.

### Requirements
1. Fetch API must be used
2. Following endpoint must be used: `https://training.nerdbord.io/api/v1/fischkapp/flashcards`

### Code Review

#### package.json
The addition of the "uuid" package is appropriate as it is used in the code to generate unique identifiers for new cards.

#### src/App.tsx
The changes in this file meet the requirements. The Fetch API is used to get data from the provided endpoint when the app initializes. The useEffect hook is used to fetch data when the app initializes and whenever a new card is added (isCardAdding changes). The structure of CardData has been updated to match the data from the server.

However, there's a redundant piece of code in the useEffect hook:

```javascript
data.forEach((card: CardData) => {
  _id: card._id;
  front: card.front;
  back: card.back;
});
```
This forEach loop doesn't do anything and can be removed.

#### src/components/AddCard/AppAddCard.tsx
The changes in this file are correct. The uuid package is used to generate a unique identifier for each new card. The structure of newCard matches the structure of CardData.

#### src/components/Card/AppCard.tsx
The changes in this file are correct. The properties of AppCardProps and the props of AppCardNormal and AppCardEdit components have been updated to match the structure of CardData.

#### src/components/Card/AppCards.tsx
The changes in this file are correct. The props of AppCard components have been updated to match the structure of CardData.

#### src/components/Card/CardEdit/AppCardEdit.tsx
The changes in this file are correct. The properties of AppCardEdtiProps have been updated to match the structure of CardData. The editCardData and deleteCardData functions now use _id instead of id.

However, the editCardData function doesn't update the card correctly. It should update the "front" and "back" properties, not "cardQuestion" and "cardAnswer". The corrected code is:

```javascript
if (card._id === _id) return {
  ...card, front: question, back: answer
};
```

#### src/components/Card/CardNormal/AppCardNormal.tsx
The changes in this file are correct. The properties of AppCardNormalProps have been updated to match the structure of CardData.

### Conclusion
The code changes mostly cover the task requirements. The Fetch API is used to get data from the correct endpoint, and the structure of CardData matches the data from the server. However, there's a redundant forEach loop in src/App.tsx and a mistake in src/components/Card/CardEdit/AppCardEdit.tsx.

**Review by NerdbordMentor — APPROVED** [review]
Based on the provided task requirements and code changes, here is the code review:

1. **package.json**: The addition of "uuid" package is appropriate as it is used in the code to generate unique identifiers for the flashcards.

2. **src/App.tsx**: The changes in this file are in line with the task requirements. The Fetch API is used to get all existing flashcards from the database when initializing the app. The useEffect hook is correctly used to fetch data from the provided endpoint when the app is loaded and when a new card is added (isCardAdding changes). The CardData interface has been updated to match the data structure returned from the API.

3. **src/components/AddCard/AppAddCard.tsx**: The changes in this file are correct. The uuid package is used to generate a unique identifier for each new card, replacing the previous method of generating an id. The newCard object now correctly matches the structure of the CardData interface.

4. **src/components/Card/AppCard.tsx**: The changes in this file are correct. The props of the AppCard component have been updated to match the structure of the CardData interface.

5. **src/components/Card/AppCards.tsx**: The changes in this file are correct. The props passed to the AppCard component have been updated to match the structure of the CardData interface.

6. **src/components/Card/CardEdit/AppCardEdit.tsx**: The changes in this file are correct. The id prop has been replaced with _id to match the structure of the CardData interface. The editCardData and deleteCardData functions have been updated to use _id instead of id.

7. **src/components/Card/CardNormal/AppCardNormal.tsx**: The changes in this file are correct. The question and answer props have been replaced with front and back to match the structure of the CardData interface.

In conclusion, all code changes cover the task requirements. The Fetch API is used to get all existing flashcards from the database when initializing the app, and the endpoint provided in the task requirements is used. The structure of the CardData interface and the related components have been updated to match the data returned from the API.
